src/lib.rs

```diff
@@ -1,101 +1,57 @@
-#![recursion_limit="128"]
+extern crate uuid;
+pub use uuid::Uuid;
 
 #[macro_use]
 extern crate proc_macro_hack;
-#[macro_use]
-extern crate syn;
-#[macro_use]
-extern crate quote;
 
+#[macro_use]
+#[allow(unused_imports)]
+extern crate uuid_macro_impl;
 
-proc_macro_expr_impl! {
-    pub fn uuid_parts_impl(input: &str) -> String {
+#[doc(hidden)]
+#[allow(unused_imports)]
+use uuid_macro_impl::*;
 
-        let mut i = 0usize;
-        let mut bytes = [0u8; 16];
+proc_macro_expr_decl! {
+    #[doc(hidden)]
+    uuid_parts! => uuid_parts_impl
+}
 
-        let hex = |m: char| {
-            match m {
-                '0' => 0u8,
-                '1' => 1u8,
-                '2' => 2u8,
-                '3' => 3u8,
-                '4' => 4u8,
-                '5' => 5u8,
-                '6' => 6u8,
-                '7' => 7u8,
-                '8' => 8u8,
-                '9' => 9u8,
-                'A' | 'a' => 10u8,
-                'B' | 'b' => 11u8,
-                'C' | 'c' => 12u8,
-                'D' | 'd' => 13u8,
-                'E' | 'e' => 14u8,
-                'F' | 'f' => 15u8,
-                _ => panic!("unexpected char.")
-            }
-        };
-        for c in input.chars() {
-            if c == '-' || c == '"' {
-                continue;
-            }
-            let index = i / 2;
-            if index >= 16 {
-                panic!("out ouf index.")
-            }
-            bytes[index] = if i % 2 == 0 {hex(c)} else { bytes[index] * 16 + hex(c) };
-            i += 1;
+/// A literal syntax for generating UUID values.
+///
+/// ```
+/// # #[macro_use]
+/// # extern crate uuid_macro;
+/// # use uuid_macro::*;
+/// # fn main() {
+/// #     let _ : Uuid = uuid!("6B29FC40-CA47-1067-B31D-00DD010662DA");
+/// # }
+/// ```
+#[macro_export]
+macro_rules! uuid {
+    {$literal:expr} => {
+        {
+            const BYTES: [u8; 16] = uuid_parts! {$literal};
+            $crate::Uuid::from_uuid_bytes(BYTES)
         }
+    }
+}
 
+#[cfg(test)]
+mod tests {
+    use super::*;
+    #[cfg(feature = "nightly")]
+    const CONST_UUID: Uuid = uuid!("F9168C5E-CEB2-4FAA-B6BF-329BF39FA1E4");
 
+    #[cfg(feature = "nightly")]
```

Comment:
just use `#[cfg(nightly)]`

Reply:
@kinggoesgaming Are you sure ```#[cfg(nightly)]```really working? I create an new hello-word project with `cargo new hello-world --bin`, and modify main.rs :
```rust
fn main() {
    test();
}

#[cfg(nightly)]
fn test() {
    println!("nightly, Hello, world!");
}

#[cfg(not(nightly))]
fn test() {
    println!("not nightly, Hello, world!");
}

```
Run with command ```cargo +nightly run```,It still prints  `not nightly, Hello, world!`;

Reply:
@kinggoesgaming yah he is right. I reproduced it [here](https://play.rust-lang.org/?version=nightly&mode=debug) and I'm getting the same output

Reply:
Something funky is definitely going on... Imma fiddle around today on this

Reply:
alright after going through the mightly google, I seems I was wrong. `#[cfg(nightly)]` was discussed at one point, however no resolution... This includes the `mod foo`

This needs resolution in a way that user does not have to explicitly provide the `nightly` feature gate. I will probably use a build script in combination with `rustc_version` for this

Reply:
@mokeyish see uuid-rs/uuid#220

I think the same setup can be added here (perhaps as a separate PR that is merged before this PR)

Reply:
@kinggoesgaming what's the issue with providing an explicit feature in order to enable an unstable `nightly` Rust feature?

Reply:
proceed with the `#[cfg(feature = "nightly")]`